code/implementation/docker/storage/src/main/java/io/cattle/platform/docker/storage/DockerImageCredentialLookup.java

```diff
@@ -1,33 +1,36 @@
 package io.cattle.platform.docker.storage;
 
+import static io.cattle.platform.core.model.tables.CredentialTable.CREDENTIAL;
+import static io.cattle.platform.core.model.tables.StoragePoolTable.STORAGE_POOL;
 import io.cattle.platform.core.constants.CommonStatesConstants;
 import io.cattle.platform.core.model.Credential;
 import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
 import io.cattle.platform.docker.client.DockerImage;
 import io.cattle.platform.storage.ImageCredentialLookup;
 import io.cattle.platform.util.type.CollectionUtils;
 
 import java.util.List;
 
-public class DockerImageCredentialLookup implements ImageCredentialLookup {
+public class DockerImageCredentialLookup extends AbstractJooqDao implements ImageCredentialLookup {
 
     private static final String REGISTRY = "registry";
     private static final String REGISTRY_CREDENTIAL = "registryCredential";
 
     @Override
-    public Credential getDefaultCredential(String uuid, List<?> storagePools, List<?> credentials) {
+    public Credential getDefaultCredential(String uuid, long currentAccount) {
         DockerImage image = DockerImage.parse(uuid);
         if ( image == null){
             return null;
         }
         String serverAddress = image.getServer();
         Long registryId = null;
-        for(Object item: storagePools){
-            if (!(item instanceof StoragePool)) {
-                continue;
-            }
-            StoragePool registry = ((StoragePool) item);
-            if (!(registry.getKind().equalsIgnoreCase(REGISTRY)) || !registry.getState().equalsIgnoreCase(CommonStatesConstants.ACTIVE)){
+        List<StoragePool> storagePools = create().selectFrom(STORAGE_POOL)
+                .where(STORAGE_POOL.STATE.eq(CommonStatesConstants.ACTIVE)
+                        .and(STORAGE_POOL.ACCOUNT_ID.eq(currentAccount)
+                        .and(STORAGE_POOL.KIND.eq("registry")))).fetchInto(StoragePool.class);
```

Comment:
Minor - we should remove the hardcoded "registry" reference and point to REFERENCE instead